monai/networks/nets/vnet.py

```diff
@@ -0,0 +1,213 @@
+        dropout_type: Type[Union[nn.Dropout, nn.Dropout2d, nn.Dropout3d]] = Dropout[Dropout.DROPOUT, dropout_dim]
+
+        self.up_conv = conv3d_trans_type(inchans, outchans // 2, kernel_size=2, stride=2)
+        self.bn1 = norm3d_type(outchans // 2)
+        self.dropout = dropout_type(dropout_p) if dropout_p is not None else None
+        self.dropout2 = dropout_type(0.5)
+        self.relu1 = get_acti_layer(elu, outchans // 2)
+        self.relu2 = get_acti_layer(elu, outchans)
+        self.ops = _make_nconv(outchans, nconvs, elu)
+
+    def forward(self, x, skipx):
+        if self.dropout is not None:
+            out = self.dropout(x)
+        else:
+            out = x
+        skipxdo = self.dropout2(skipx)
+        out = self.relu1(self.bn1(self.up_conv(out)))
+        xcat = torch.cat((out, skipxdo), 1)
+        out = self.ops(xcat)
+        out = self.relu2(torch.add(out, xcat))
+        return out
+
+
+class OutputTransition(nn.Module):
+    def __init__(self, inchans: int, outchans: int, elu: bool):
+        super(OutputTransition, self).__init__()
+
+        conv3d_type: Type[nn.Conv3d] = Conv[Conv.CONV, 3]
+        act = ("elu", {"inplace": True}) if elu else ("prelu", {"num_parameters": outchans})
+        self.conv_block = Convolution(
+            dimensions=3, in_channels=inchans, out_channels=outchans, kernel_size=5, act=act, norm=Norm.BATCH,
+        )
+        self.conv2 = conv3d_type(outchans, outchans, kernel_size=1)
+
+    def forward(self, x):
+        # convolve 32 down to 2 channels
+        out = self.conv_block(x)
+        out = self.conv2(out)
+        return out
+
+
+class VNet(nn.Module):
+    """
+    V-Net: Fully Convolutional Neural Networks for Volumetric Medical Image Segmentation
+    The official Caffe implementation is available in the faustomilletari/VNet repo on GitHub:
+    https://github.com/faustomilletari/VNet
+    The code is adapted from:
+    https://github.com/mattmacy/vnet.pytorch/blob/master/vnet.py
+
+    Args:
+        num_input_channel: number of the input image channel. Defaults to 1. The value should meet
+            the condition that: 16 % num_input_channel == 0.
+        num_output_channel: number of the output image channel. Defaults to 1.
+        elu: use ELU activation or PReLU activation. Defaults to False, means using PReLU.
+        n_spatial_dim: spatial dimension of input data.
+        dropout_p: dropout ratio. Defaults to 0.5.
+        dropout_dim: determine the dimensions of dropout. Defaults to 3.
+            When dropout_dim = 1, randomly zeroes some of the elements for each channel.
+            When dropout_dim = 2, Randomly zero out entire channels (a channel is a 2D feature map).
+            When dropout_dim = 3, Randomly zero out entire channels (a channel is a 3D feature map).
+    """
+
+    def __init__(
+        self,
+        num_input_channel: int = 1,
+        num_output_channel: int = 1,
+        elu: bool = False,
```

Comment:
could make this a generic `act` parameter defaults to elu, something like: `act: Union[Tuple[str, Dict], str] = ("elu", {"inplace": True}),`

Reply:
I see, I've changed them.